### PR TITLE
Add logging for state sync errors

### PR DESCRIPTION
**Pull request by mikhailOK**

### CODE REVIEW

File: chain/client/src/client_actor.rs [hunk 1]
```diff
@@ -339,13 +339,13 @@ impl Handler<NetworkClientMessages> for ClientActor {
                 }
 
                 for shards_to_download in shards_to_download_vec {
-                    let shard_sync_download = if shards_to_download.contains_key(&shard_id) {
-                        &shards_to_download[&shard_id]
-                    } else {
-                        // TODO is this correct behavior?
-                        continue;
+                    let shard_sync_download = match shards_to_download.get_mut(&shard_id) {
```
Comment (Kouprin) [inline chain/client/src/client_actor.rs]: does it work well? I'm surprised

File: chain/client/src/client_actor.rs [hunk 2]
```diff
@@ -380,10 +381,11 @@ impl Handler<NetworkClientMessages> for ClientActor {
                                         part.clone(),
                                     ) {
                                         Ok(()) => {
-                                            new_shard_download.downloads[part_id].done = true;
+                                            shard_sync_download.downloads[part_id].done = true;
                                         }
-                                        Err(_err) => {
-                                            new_shard_download.downloads[part_id].error = true;
+                                        Err(err) => {
+                                            error!(target: "sync", "Error with state part {} for shard {}: {:?}", part_id, shard_id, err);
```
Comment (Kouprin) [inline chain/client/src/client_actor.rs]: `error!(target: "sync", "State sync error, shard = {}, part = {}, hash = {}: {:?}", shard_id, part_id, hash, err);`

File: chain/client/src/client_actor.rs [hunk 3]
```diff
@@ -357,10 +357,11 @@ impl Handler<NetworkClientMessages> for ClientActor {
                                         header.clone(),
                                     ) {
                                         Ok(()) => {
-                                            new_shard_download.downloads[0].done = true;
+                                            shard_sync_download.downloads[0].done = true;
                                         }
-                                        Err(_err) => {
-                                            new_shard_download.downloads[0].error = true;
+                                        Err(err) => {
+                                            error!(target: "sync", "Error with state header for shard {}: {:?}", shard_id, err);
```
Comment (Kouprin) [inline chain/client/src/client_actor.rs]: `error!(target: "sync", "State sync header error, shard = {}, hash = {}: {:?}", shard_id, hash, err);`

File: chain/client/src/sync.rs [hunk 4]
```diff
@@ -590,16 +587,17 @@ impl StateSync {
...
+                            *shard_sync_download = ShardSyncDownload {
                                 downloads: vec![],
                                 status: ShardSyncStatus::StateDownloadComplete,
                             }
                         }
-                        _ => {
+                        Err(e) => {
                             // Cannot finalize the downloaded state.
                             // The reasonable behavior here is to start from the very beginning.
+                            error!(target: "sync", "Error finalizing state download: {:?}", e);
```
Comment (Kouprin) [inline chain/client/src/sync.rs]: error!(target: "sync", "State sync finalizing error, shard = {}, hash = {}: {:?}", shard_id, sync_hash, e);

File: chain/client/src/client_actor.rs [hunk 5]
```diff
@@ -348,13 +348,13 @@ impl Handler<NetworkClientMessages> for ClientActor {
                 }
 
                 for shards_to_download in shards_to_download_vec {
```
Comment (ilblackdragon) [inline chain/client/src/client_actor.rs]: this vec seems error prone, suggestion to either keep only either statesync and state_sync_infos.
We can merge this hashmaps and assert that there is no common keys.
and then if there is missing `shard_id` in this hashmap - we should show the error / possibly ban the peer.

**Pull request merged**